RFS/src/main/java/org/opensearch/migrations/bulkload/transformers/TransformFunctions.java

```diff
@@ -16,27 +16,108 @@ private TransformFunctions() {}
 
     /* Turn dotted index settings into a tree, will start like:
      * {"index.number_of_replicas":"1","index.number_of_shards":"5","index.version.created":"6082499"}
+     * 
+     * Special handling for conflicting keys: When a scalar value like "knn": "true" exists
+     * alongside nested properties like "knn.space_type": "l2", we keep the settings flat
+     * to preserve both values. This means the output will have both "knn": "true" and
+     * "knn.space_type": "l2" as separate flat keys rather than nesting space_type under knn.
```

Comment:
Adding 2-3 more sentences describing an overview of the strategy would make it much easier for the reader to understand what you're trying to achieve.